x-pack/plugins/enterprise_search/public/applications/enterprise_search/components/trial_callout/trial_callout.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues } from 'kea';
+import moment from 'moment';
+
+import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { LicensingLogic } from '../../../shared/licensing';
+
+export const TrialCallout: React.FC = () => {
+  const { license, isTrial } = useValues(LicensingLogic);
+
+  if (!isTrial) return null;
+
+  const title = (
+    <FormattedMessage
+      id="xpack.enterpriseSearch.trialCalloutTitle"
+      defaultMessage="Your Elastic Stack Trial license, which enables Platinum features, expires in {days, plural, one {# day} other {# days}}. {link}"
+      values={{
+        days: moment(license?.expiryDateInMillis).diff(moment({ hours: 0 }), 'days'),
+        link: (
+          <EuiLink href="https://www.elastic.co/subscriptions" target="_blank">
+            <strong>Learn more about Elastic Stack licenses.</strong>
+          </EuiLink>
+        ),
+      }}
+    />
+  );
+
+  return (
+    <>
+      <EuiCallOut size="s" title={title} iconType="iInCircle" />
```

Comment:
[Optional] Not sure if you care, but we can quickly/hackily-ish get this centered with some inline styles if you want to match the `ent-search` layout a bit more closely:

```suggestion
      <EuiCallOut size="s" title={title} iconType="iInCircle" style={{ margin: '0 auto' }} />
```

Before:
<img width="1326" alt="" src="https://user-images.githubusercontent.com/549407/117052123-35273580-accc-11eb-9337-830cabe81aed.png">

After:
<img width="1078" alt="" src="https://user-images.githubusercontent.com/549407/117052139-39ebe980-accc-11eb-8d47-16d685596094.png">